### PR TITLE
feat(wallet): Asset Item More Menu

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
- Adds a `More Menu` to `Asset Items` on the `Portfolio` page and `Account Details` page
- `More Menu` includes the following options if available for that asset (Buy, Send, Swap, Deposit, Sell)
- Fixes a bug where the `Fund` screens were not auto filling the `Search Value` to the `Selected Asset` from params.

Resolves <https://github.com/brave/brave-browser/issues/30982>
Resolves <https://github.com/brave/brave-browser/issues/32026>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Go to the `Portfolio` screen and click on the `More Menu` for an asset
2. Test all options and make sure they work properly
3. `Buy` button will be available if that `Asset` is `Buyable`
4. `Send` button will be available if that `Asset` has a balance
5. `Swap` button will be available if that `Asset` has a balance
6. `Deposit` button should always be available
7. `Sell` button should be available if that `Asset` has a balance and your `Portfolio` is grouped by `Accounts`

https://github.com/brave/brave-core/assets/40611140/422dee25-7aa6-4037-9bfe-86f12ec22902

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/wallet-menus/asset-item-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    ? {
+      accounts: [account],
+      networks: networkList
+    }
+    : skipToken
+  )
+
+  // Hooks
+  const {
+    selectedSellAsset,
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/wallet-menus/asset-item-menu.tsx:85]: Is `selectedSellAsset` ever different from the `asset` prop? If no, the `useBalancesFetcher` is not necessary.

**Review by onyb — COMMENTED** [review]
Looks good overall, but there's an android browser test failing for a legit reason. Also left one small remark.

File: components/brave_wallet_ui/common/slices/endpoints/off-ramp.endpoints.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        }
+      }, providesTags: (results, error, _arg) => {
+        if (error) {
+          return ['UNKNOWN_ERROR']
+        }
+        return ['OffRampAssets']
+      }
+    })
+  }
+}
```
Comment (muliswilliam) [inline components/brave_wallet_ui/common/slices/endpoints/off-ramp.endpoints.ts:103]: nit: Missing new line at end of file

File: components/brave_wallet_ui/common/slices/endpoints/on-ramp.endpoints.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            data: results
+          }
+        } catch (error) {
+          const errorMessage = `Unable to fetch onRamp assets: ${error}`
+          console.log(errorMessage)
+          return {
+            error: errorMessage
+          }
+        }
+      }, providesTags: (results, error, _arg) => {
```
Comment (muliswilliam) [inline components/brave_wallet_ui/common/slices/endpoints/on-ramp.endpoints.ts:198]: Results is not used here
```suggestion
      }, providesTags: (_results, error, _arg) => {
```

**Review by muliswilliam — CHANGES_REQUESTED** [review]
Looks great. Requested some changes

File: components/brave_wallet_ui/common/slices/endpoints/off-ramp.endpoints.ts (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/slices/endpoints/off-ramp.endpoints.ts:103]: It actually is there `Github` just doesn't show it 

Screenshot 39

If it were missing `Github` would display an error like so

Screenshot 42

File: components/brave_wallet_ui/common/slices/endpoints/on-ramp.endpoints.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        if (error) {
+          return ['UNKNOWN_ERROR']
+        }
+        return ['OnRampAssets']
+      }
+    })
+  }
+}
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/slices/endpoints/on-ramp.endpoints.ts:206]: Same

Screenshot 40

**Review by muliswilliam — APPROVED** [review]
LGTM

File: components/brave_wallet_ui/components/desktop/wallet-menus/asset-item-menu.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/wallet-menus/asset-item-menu.tsx:69]: Good catch, removed and is now directly passing the `assetBalance` prop!

**Review by onyb — APPROVED** [review]
In case the Android test fails again, I shared the fix on Slack. Looks great otherwise!

**Review by darkdh — APPROVED** [review]

File: components/brave_wallet/browser/android_page_appearing_browsertest.cc [hunk 5]
```diff
@@ -425,14 +425,16 @@ IN_PROC_BROWSER_TEST_F(AndroidPageAppearingBrowserTest,
                        TestDepositPageAppearing) {
   GURL url = GURL("chrome://wallet/deposit-funds");
   const std::vector<std::string> ignore_patterns = {
-      "TypeError: Cannot read properties of undefined (reading 'forEach')"};
+      "TypeError: Cannot read properties of undefined (reading 'forEach')",
+      "TypeError: Cannot read properties of undefined (reading 'match')"};
```
Comment (simoarpe) [inline components/brave_wallet/browser/android_page_appearing_browsertest.cc:429]: Adding this line breaks the test, as it now passes but the deposit page does not load correctly.

Screenshot 2023-08-03 at 4 22 16 PM

**Pull request merged**